dartagnan/src/main/java/com/dat3m/dartagnan/program/event/core/Alloc.java

```diff
@@ -103,4 +121,16 @@ protected String defaultString() {
 
     @Override
     public Alloc getCopy() { return new Alloc(this); }
+
+    @Override
+    public <T> T accept(EventVisitor<T> visitor) {
+        return visitor.visitAlloc(this);
+    }
+
+    @Override
+    public BooleanFormula encodeExec(EncodingContext ctx) {
+        return ctx.getBooleanFormulaManager().and(
+                super.encodeExec(ctx),
+                ctx.equal(ctx.result(this), ctx.encodeExpressionAt(allocatedObject, this)));
```

Comment:
Shouldn't you use `getAllocatedObject()` instead of `allocatedObject` to be sure this is not null?

Reply:
I can do, but I guess both versions result in an exception either way.